gcloud/bigtable/cluster.py

```diff
@@ -15,9 +15,51 @@
 """User friendly container for Google Cloud Bigtable Cluster."""
 
 
+import re
+
 from gcloud.bigtable.table import Table
 
 
+_CLUSTER_NAME_RE = re.compile(r'^projects/(?P<project>[^/]+)/'
+                              r'zones/(?P<zone>[^/]+)/clusters/'
+                              r'(?P<cluster_id>[a-z][-a-z0-9]*)$')
+
+
+def _require_pb_property(message_pb, property_name, value):
+    """Check that a property agrees with the value on the message.
+
+    :type message_pb: :class:`google.protobuf.message.Message`
+    :param message_pb: The message to check for ``property_name``.
+
+    :type property_name: str
+    :param property_name: The property value to check against.
+
+    :type value: object or :data:`NoneType <types.NoneType>`
```

Comment:
@tseaver Note this looks like YAGNI, but this has been pulled in from the "future" and is used to make sure values are set / correct on a PB object.